src/main/java/app/onepass/organizer/services/EventService.java

```diff
@@ -0,0 +1,287 @@
+package app.onepass.organizer.services;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import app.onepass.apis.CreateEventRequest;
+import app.onepass.apis.Duration;
+import app.onepass.apis.Event;
+import app.onepass.apis.GetByIdRequest;
+import app.onepass.apis.GetEventByIdResponse;
+import app.onepass.apis.GetEventResponse;
+import app.onepass.apis.HasEventRequest;
+import app.onepass.apis.OrganizerServiceGrpc;
+import app.onepass.apis.RemoveEventRequest;
+import app.onepass.apis.Result;
+import app.onepass.apis.UpdateEventDurationRequest;
+import app.onepass.apis.UpdateEventFacilityRequest;
+import app.onepass.apis.UpdateEventInfoRequest;
+import app.onepass.apis.UpdateRegistrationRequestRequest;
+import app.onepass.apis.UserRequest;
+import app.onepass.organizer.entities.EventDurationEntity;
+import app.onepass.organizer.entities.EventEntity;
+import app.onepass.organizer.entities.EventRegistrationEntity;
+import app.onepass.organizer.messages.EventMessage;
+import app.onepass.organizer.messages.FacilityMessage;
+import app.onepass.organizer.repositories.EventDurationRepository;
+import app.onepass.organizer.repositories.EventRegistrationRepository;
+import app.onepass.organizer.repositories.EventRepository;
+import app.onepass.organizer.repositories.FacilityRepository;
+import app.onepass.organizer.utilities.ServiceUtil;
+import app.onepass.organizer.utilities.TimeUtil;
+import io.grpc.stub.StreamObserver;
+
+@Service
+public class EventService extends OrganizerServiceGrpc.OrganizerServiceImplBase {
+
+	@Autowired
+	private EventRepository eventRepository;
+
+	@Autowired
+	private FacilityRepository facilityRepository;
+
+	@Autowired
+	private EventDurationRepository eventDurationRepository;
+
+	@Autowired
+	private EventRegistrationRepository eventRegistrationRepository;
+
+	@Override
+	@Transactional
+	public void createEvent(CreateEventRequest request, StreamObserver<Result> responseObserver) {
+
+		if (eventRepository.findById(request.getEvent().getId()).isPresent()) {
+
+			Result result = ServiceUtil.returnError("An event with this ID already exists.");
```

Comment:
Should we custom error as new class?

Reply:
return error is not an error class, it’s a method for returning an error result of isok = false